// vue-responsiveness/dist/vue-responsiveness.js
import { reactive as p, computed as o } from "vue";
const f = {
  Bootstrap_3: {
    xs: 0,
    sm: 576,
    lg: 992
  },
  Bootstrap_4: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  },
  Bootstrap_5: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200,
    xxl: 1400
  },
  Bulma: {
    mobile: 0,
    tablet: 769,
    desktop: 1024,
    widescreen: 1216,
    fullhd: 1408
  },
  Chakra: {
    xs: 0,
    sm: 480,
    md: 768,
    lg: 992,
    xl: 1280,
    "2xl": 1472
  },
  Foundation: {
    small: 0,
    medium: 640,
    large: 1024,
    xlarge: 1200,
    xxlarge: 1440
  },
  Ionic: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  },
  Material_Design: {
    xs: 0,
    sm_8: 600,
    sm_12: 905,
    md: 1240,
    lg: 1440
  },
  Materialize: {
    s: 0,
    m: 600,
    l: 992,
    xl: 1200
  },
  Material_UI: {
    xs: 0,
    sm: 600,
    md: 960,
    lg: 1260,
    xl: 1920
  },
  Quasar: {
    xs: 0,
    sm: 600,
    md: 1024,
    lg: 1440,
    xl: 1920
  },
  Semantic_UI: {
    mobile: 0,
    tablet: 768,
    small_monitor: 992,
    large_monitor: 1200
  },
  Skeleton: {
    mobile: 0,
    phablet: 400,
    tablet: 550,
    desktop: 750,
    desktop_hd: 1e3,
    desktop_hd_lg: 1200
  },
  Tailwind_CSS: {
    xs: 0,
    sm: 640,
    md: 768,
    lg: 1024,
    xl: 1280,
    "2xl": 1536
  },
  Vuetify: {
    xs: 0,
    sm: 600,
    md: 960,
    lg: 1280,
    xl: 1920
  },
  Windi_CSS: {
    xs: 0,
    sm: 640,
    md: 768,
    lg: 1024,
    xl: 1280,
    "2xl": 1536
  }
};
let t;
const _ = {
  install(r, c = f.Bootstrap_5) {
    if (typeof window < "u") {
      const i = Object.entries(c).sort(([, e], [, s]) => (e || 0) - (s || 0)).reduce((e, [s, m], n, l) => {
        var a;
        return e[s] = {
          min: m ? `(min-width: ${m}px)` : "",
          max: (a = l[n + 1]) != null && a[1] ? `(max-width: ${l[n + 1][1] - 0.1}px)` : ""
        }, e;
      }, {});
      t = p({
        isMin: o(() => (e) => {
          var s;
          return ((s = t[e]) == null ? void 0 : s.min) || !1;
        }),
        isMax: o(() => (e) => {
          var s;
          return ((s = t[e]) == null ? void 0 : s.max) || !1;
        }),
        isOnly: o(() => (e) => {
          var s;
          return ((s = t[e]) == null ? void 0 : s.only) || !1;
        }),
        ...Object.assign(
          {},
          ...Object.keys(i).map((e) => ({
            [e]: { min: !1, max: !1, only: !1 }
          })),
          {
            current: o(
              () => {
                var e;
                return (e = Object.entries(t).find(
                  ([, s]) => typeof s == "object" && s.only
                )) == null ? void 0 : e[0];
              }
            )
          }
        )
      }), Object.entries(i).forEach(([e, s]) => {
        const m = {
          min: window.matchMedia(s.min),
          max: window.matchMedia(s.max)
        };
        Object.entries(m).forEach(([n, l]) => {
          const a = ({ matches: g }) => {
            const { min: x, max: d } = { ...t[e], [n]: g };
            t[e] = { min: x, max: d, only: x && d };
          };
          l.addEventListener("change", a), a(l);
        });
      }), r.config.globalProperties.$matches = t;
    }
  }
}, b = () => t;
export {
  f as Presets,
  _ as VueResponsiveness,
  b as useMatches
};
